vyper/builtin_functions/functions.py

```diff
@@ -699,25 +700,19 @@ def evaluate(self, node):
 
         if node.keywords:
             return_type = get_type_from_annotation(node.keywords[0].value, DataLocation.UNSET)
-            if isinstance(return_type, Bytes32Definition):
-                length = 32
-            elif isinstance(return_type, BytesArrayDefinition) and return_type.length == 4:
-                length = 4
-            else:
-                raise ArgumentException("output_type must be bytes[4] or bytes32", node.keywords[0])
+            if not isinstance(return_type, Bytes4Definition) and not (
```

Comment:
this check is a little hard to read. maybe it would be cleaner if we preserved the original structure, e.g.

```python
if isinstance(return_type, Bytes4Definition):
    is_bytes4 = True
elif isinstance(return_type, BytesArrayDefinition) and return_type.length == 4:
    is_bytes4 = False
else:
    raise ArgumentException(...)
```

Reply:
I skipped the "else" branch by setting the default before the "if" branch, but not sure if this impacts readability still?

Reply:
i think the original was clearest, it looks a bit strange to have an exception raised in a "dangling" `elif` branch. also on line 712, please simplify to
```

        if is_bytes4:
            return vy_ast.Hex.from_node(node, value=hex(value))
        else:
            return vy_ast.Bytes.from_node(node, value=value.to_bytes(4, "big"))
```